otherlibs/win32unix/stat.c

```diff
@@ -117,23 +117,13 @@ static value stat_aux(int use_64, __int64 st_ino, struct _stat64 *buf)
 
 static int convert_time(FILETIME* time, __time64_t* result, __time64_t def)
 {
-  SYSTEMTIME sys;
-  FILETIME local;
+  unsigned __int64 utime = *((unsigned __int64*)time);
 
-  if (time->dwLowDateTime || time->dwHighDateTime) {
-    if (!FileTimeToLocalFileTime(time, &local) ||
-        !FileTimeToSystemTime(&local, &sys))
-    {
-      win32_maperr(GetLastError());
-      return 0;
-    }
-    else
-    {
-      struct tm stamp = {sys.wSecond, sys.wMinute, sys.wHour,
-                         sys.wDay, sys.wMonth - 1, sys.wYear - 1900,
-                         0, 0, 0};
-      *result = _mktime64(&stamp);
-    }
+  if (utime) {
+    /* There are 11644473600000 seconds since 1 January 1601 (the NT Epoch) and
+     * 1 January 1970 (the Unix Epoch). FILETIME is measured in 100ns ticks.
+     */
+    *result = (utime - 116444736000000000ULL);
```

Comment:
Another good piece :blush:

A museum curator did this before elsewhere: https://github.com/ocaml/ocaml/blame/trunk/byterun/ints.c#L504-L510